### PR TITLE
Updated gamesUtil.js (Closes #34)

### LINKED ISSUE #34: Sorted list of recent games
**Describe the feature you would like**
The list of games in the members area is sorted kind of random due to loading latency.
It would be nice if the newest game would always be at the first position.

There is a pull request fixing this issue.

### PR DESCRIPTION
**Pull request by vincentscode**
Modified so the games will always be sorted correctly.
(I had to make a new pull request due to merge conflicts.)

## Description
There is no need for an async task inside in async task. It destroyes the order of the recent games.

## Related Issue
#34

## Motivation and Context
The change solves the problem of always having to search for recent games in the statistics.

## How Has This Been Tested?
This change has been tested locally with an https-server.

## Screenshots (if appropriate):
Unnecessary.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Last time I already said that using `async: false` is a bad idea and you ( @VincentsCode ) said

> I see no point in just calling something bad for no reason.
It works well and is easy to understand. That are the things that count.

Because my explanation was clearly not clear enough,
I am going to explain it again, why that is an extremely bad idea:
A view on what is on the Developer Console of your favorite browser when
running an ajax call without enabled async, you are very likely to see
something like:

`[Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.`

I don't go into detail here, because I hope that you are able to read for yourself.


A way better solution is recursive async calls, because it does not have
the negative effects of synchronous XMLHttpRequest.

Snippet of a working async implementation:
```javascript
function display_all_my_games(userid) {
    let my_url = get_games_url() + 'api/games/?user_ids=' + userid;
    $.ajax({
        url: my_url,
        type: "get",
        contentType: "application/json; charset=utf-8",
        cache: false,
        success: function (data, status, xhr) {
            display_game(data.games.map(x => x.id));
        }
    });
}

function display_game(game_ids) {
    let current = game_ids.shift(-1);
    let my_url = get_games_url() + 'api/games/' + current;
    $.ajax({
        url: my_url,
        type: "get",
        contentType: "application/json; charset=utf-8",
        cache: false,
        success: function (data, status, xhr) {
            console.log(current, data); //TODO do something with the data
            if (game_ids.length > 0) display_game(game_ids);
        }
    });
}
```

Don't say someone calls something bad for no reason... for no reason.

If you want to make this even better, you could add a limit parameter to the 
backend with an page support, because just imagine someone having like 1000+ games.

The argument that this is recursive and for that reason it's memory intensive
etc. is not real and if you care about memory saving, read the paragraph over this one.

**Comment by Scriptim** [comment]
> (I had to make a new pull request due to merge conflicts.)

You didn't have to.

All you had to do was to merge the master branch into your branch. On GitHub this is even possible [via the interface](https://help.github.com/articles/resolving-a-merge-conflict-on-github/ "Resolving a merge conflict on GitHub").

**Comment by ghost** [comment]
@VincentsCode So could you please edit the JS according to my "instructions"?

**Comment by vincentscode** [comment]
> > (I had to make a new pull request due to merge conflicts.)
> 
> You didn't have to.
> 
> All you had to do was to merge the master branch into your branch. On GitHub this is even possible [via the interface](https://help.github.com/articles/resolving-a-merge-conflict-on-github/).

Yeah, i kind of screwed up and had to hard reset the branch. 



> @VincentsCode So could you please edit the JS according to my "instructions"?

Done , thank you for the hint.

File: FrontPage/center/apis/games/games.js [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+function display_game(game_ids) {
+    let current = game_ids.shift(-1);
+    let my_url = get_games_url() + 'api/games/' + current;
+    $.ajax({
+        url: my_url,
+        type: "get",
+        contentType: "application/json; charset=utf-8",
+        cache: false,
+        success: function (data, status, xhr) {
+            console.log(current, data); //TODO do something with the data
```
Comment (ghost) [inline FrontPage/center/apis/games/games.js]: There is a // TODO with a reason.
I think its pretty self explanatory. :grin:

**Pull request merged**